Sources/PrimerSDK/Classes/PCI/Services/DefaultNetworkService.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  DefaultNetworkService.swift
+//  PrimerSDK
+//
+//  Created by Jack Newcombe on 07/03/2024.
+//
+
+import Foundation
+
+protocol ResponseMetadata {
+    var responseUrl: String? { get }
+    var statusCode: Int { get }
+    var headers: [String: String]? { get }
+}
+
+extension ResponseMetadata {
+    var description: String {
+        "\(responseUrl ?? "Unknown URL") => \(statusCode)"
+    }
+}
+
+extension HTTPURLResponse: ResponseMetadata {
+
+    var responseUrl: String? {
+        return url?.absoluteString
```

Comment:
Nit-pick: we could omit this return and the next. Already the case on line 18. Small thing, but if we were to choose one or the other, I vote omitting `return` where we can